src/activity-window-host.js

```diff
@@ -351,9 +358,19 @@ export class ActivityWindowRedirectHost {
       this.requestId_ = request.requestId;
       this.args_ = request.args;
       this.returnUrl_ = request.returnUrl;
-      this.targetOrigin_ = getOriginFromUrl(request.returnUrl);
-      // TODO(dvoytenko): Use `document.referrer` to verify origin.
-      this.targetOriginVerified_ = false;
+      if (request.origin) {
+        // Trusted request: trust origin and verified flag explicitly.
```

Comment:
What is the definition of trusted being used here?

Reply:
The host could get the connection parameters in a trusted way. E.g. _not_ from URL.